### PR TITLE
Sign transaction with available wallet

### LINKED ISSUE #4814: Add Use of Local Wallet Support to Non-`web3-eth-account` Packages
Various packages (namely `web3-eth`) need to check whether a local wallet has been instantiated so a `transaction.from` can be defaulted and a private key can be accessed to sign the transaction e.g.:

- [x] [One](https://github.com/ChainSafe/web3.js/blob/96026ec5990cd01a6ccfe33daa68b67de50bd1ad/packages/web3-eth/src/rpc_method_wrappers.ts#L385)
- [x] [Two](https://github.com/ChainSafe/web3.js/blob/96026ec5990cd01a6ccfe33daa68b67de50bd1ad/packages/web3-eth/src/rpc_method_wrappers.ts#L459)

### PR DESCRIPTION
**Pull request by nazarhussain**
## Description

Use wallet support to sign transaction. 

Fixes #4814

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [x] I ran `npm run test:unit` with success.
- [x] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [x] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [x] I have tested my code on the live network.
- [x] I have checked the Deploy Preview and it looks correct.
- [x] I have updated the `CHANGELOG.md` file in the root folder.

### CODE REVIEW

**Review by Spacesai1or — COMMENTED** [review]
I think this looks good so far, we just need to add support for passing an index for `address` in `sendTransaction`

**Review by avkos — APPROVED** [review]

File: packages/web3-common/src/web3_base_wallet.ts [hunk 1]
```diff
@@ -4,15 +4,32 @@ export interface Web3BaseWalletAccount {
...
 	readonly address: string;
 	readonly privateKey: string;
-	readonly signTransaction: (tx: Record<string, unknown>) => string;
-	readonly sign: (data: Record<string, unknown> | string) => string;
-	readonly encrypt: (password: string, options?: Record<string, unknown>) => Web3EncryptedWallet;
+	readonly signTransaction: (tx: Record<string, unknown>) => {
+		readonly messageHash: string;
+		readonly r: string;
+		readonly s: string;
+		readonly v: string;
```
Comment (jdevcs) [inline packages/web3-common/src/web3_base_wallet.ts:11]: Good to create type and reuse that in sign function return type as well.

Comment (jdevcs) [inline packages/web3-common/src/web3_base_wallet.ts]: Good to create type and reuse that in sign function return type as well.

**Review by jdevcs — DISMISSED** [review]

File: packages/web3-common/src/web3_base_wallet.ts (hunk 1 above)
Comment (Spacesai1or) [reply packages/web3-common/src/web3_base_wallet.ts]: Could use [HexString](https://github.com/ChainSafe/web3.js/blob/bfae214735803e38ba6bddabcf43f241d715c5d2/packages/web3-utils/src/types.ts#L1) from `web3-utils` as it's already being imported in this package

**Review by Spacesai1or — APPROVED** [review]

**Pull request merged**